In [1]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

/kaggle/input/yeh-concret-data/Concrete_Data_Yeh.csv


<font style = 'font-size: 15px;'>In this, we have to predict the strength of concrete.</font><br>
<font style = 'font-size: 15px;'>Hypothesis 0: The variables have no effect on the concrete strength.</font><br>

In [2]:
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
import statsmodels.api as sm

In [3]:
dataset = pd.read_csv('../input/yeh-concret-data/Concrete_Data_Yeh.csv')
X = dataset.iloc[:, :-1].values
y = dataset.iloc[:, -1].values

In [4]:
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.2, random_state = 0)

In [5]:
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train, y_train)

LinearRegression()

In [6]:
print(regressor.intercept_)
print(regressor.coef_)

-30.566755756266936
[ 0.11923772  0.10881555  0.0911555  -0.14527714  0.31551104  0.02225423
  0.02248514  0.11520355]


In [7]:
y_pred = regressor.predict(X_test)

In [8]:
from sklearn.metrics import r2_score
r2_score(y_test, y_pred)

0.6369606518338295

In [9]:
model_lin = sm.OLS.from_formula("csMPa ~ cement + slag + flyash + water + superplasticizer + coarseaggregate + fineaggregate + age", data=dataset)
result_lin = model_lin.fit()
result_lin.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                  csMPa   R-squared:                       0.616
Model:                            OLS   Adj. R-squared:                  0.613
Method:                 Least Squares   F-statistic:                     204.3
Date:                Mon, 28 Jun 2021   Prob (F-statistic):          6.29e-206
Time:                        19:30:18   Log-Likelihood:                -3869.0
No. Observations:                1030   AIC:                             7756.
Df Residuals:                    1021   BIC:                             7800.
Df Model:                           8                                         
Covariance Type:            nonrobust                                         
====================================================================================
                       coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------------
Intercept          -23.3312     26.586     -0.878      0.380     -75.500      28.837
cement               0.1198      0.008     14.113      0.000       0.103       0.136
slag                 0.1039      0.010     10.247      0.000       0.084       0.124
flyash               0.0879      0.013      6.988      0.000       0.063       0.113
water               -0.1499      0.040     -3.731      0.000      -0.229      -0.071
superplasticizer     0.2922      0.093      3.128      0.002       0.109       0.476
coarseaggregate      0.0181      0.009      1.926      0.054      -0.000       0.037
fineaggregate        0.0202      0.011      1.887      0.059      -0.001       0.041
age                  0.1142      0.005     21.046      0.000       0.104       0.125
==============================================================================
Omnibus:                        5.378   Durbin-Watson:                   1.282
Prob(Omnibus):                  0.068   Jarque-Bera (JB):                5.304
Skew:                          -0.174   Prob(JB):                       0.0705
Kurtosis:                       3.045   Cond. No.                     1.06e+05
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 1.06e+05. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

In [10]:
print(result_lin.pvalues)

Intercept           3.803719e-01
cement              1.897989e-41
slag                1.598993e-23
flyash              5.019648e-12
water               2.008798e-04
superplasticizer    1.810241e-03
coarseaggregate     5.442481e-02
fineaggregate       5.949075e-02
age                 5.782491e-82
dtype: float64


<font style = 'font-size: 15px;'>-> From the above data, in reference to the p-values(P > |t|), coarseaggregate(0.054) and fineaggregate(0.059) have values greater than 0.05 which means they are statistically insignificant as they are closer to proving the null hypothesis true.</font><br>
<font style = 'font-size: 15px;'>-> So, they are not considered in predicting the values for this model.</font>

In [11]:
cc = dataset[["csMPa", "cement", "slag", "flyash", "water", "superplasticizer", "coarseaggregate", "fineaggregate", "age"]].corr()
print(cc)

                     csMPa    cement      slag    flyash     water  \
csMPa             1.000000  0.497832  0.134829 -0.105755 -0.289633   
cement            0.497832  1.000000 -0.275216 -0.397467 -0.081587   
slag              0.134829 -0.275216  1.000000 -0.323580  0.107252   
flyash           -0.105755 -0.397467 -0.323580  1.000000 -0.256984   
water            -0.289633 -0.081587  0.107252 -0.256984  1.000000   
superplasticizer  0.366079  0.092386  0.043270  0.377503 -0.657533   
coarseaggregate  -0.164935 -0.109349 -0.283999 -0.009961 -0.182294   
fineaggregate    -0.167241 -0.222718 -0.281603  0.079108 -0.450661   
age               0.328873  0.081946 -0.044246 -0.154371  0.277618   

                  superplasticizer  coarseaggregate  fineaggregate       age  
csMPa                     0.366079        -0.164935      -0.167241  0.328873  
cement                    0.092386        -0.109349      -0.222718  0.081946  
slag                      0.043270        -0.283999      -0.28

<font style = 'font-size: 15px;'>-> Flyash and water have a negative relationship with the dependent variable which means that the csMPa will decrease with each 10.58% and 29% increase resp. in the independent variables.</font><br>
<font style = 'font-size: 15px;'>-> This implies that water seems to decrease the strength of the concrete as more addition of water makes the concrete dilute.</font><br>
<font style = 'font-size: 15px;'>-> Flyash can decrease the concrete strength because of:
a. Slower strength gain
b. Seasonal limitation
c. Increased need for air-entraining admixtures
d. Increase of salt scaling produced by higher proportions of fly ash
</font><br>
<font style = 'font-size: 15px;'>-> The one impacting the concrete strength the most is cement, which is 49% and because of this cement is considered one of the major ingredients in concrete.</font>